core/src/test/scala/integration/kafka/server/FetchFromFollowerIntegrationTest.scala

```diff
@@ -103,44 +105,89 @@ class FetchFromFollowerIntegrationTest extends BaseFetchRequestTest {
 
     TestUtils.generateAndProduceMessages(brokers, topic, numMessages = 10)
 
+    assertEquals(1, getPreferredReplica)
```

Comment:
For the new unit test I added a function to get preferred replica, so all explicit code to build and send fetch request got moved into the new function.